kratos/includes/model_part.h

```diff
@@ -515,6 +515,11 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
                 << "\" to the model part with name \"" << this->Name() << "\" which is not empty" << std::endl;
 
             mpVariablesList->Add(ThisVariable);
+
+            for(auto it = this->NodesBegin(); it!= this->NodesEnd(); ++it)
```

Comment:
why is this needed here?
Isn't this handled somewhere else?

Reply:
Agree with that, please wait until we discuss this in more detail.

Reply:
I tend to agree about this. Calling Merge on the variable lists after AddVariables is a better solution. However if we add a variable, and the variable is not in the node list it is a potentially very big problem